src/theory/arith/constraint.cpp

```diff
@@ -526,6 +526,50 @@ bool Constraint::hasTrichotomyProof() const {
   return getProofType() == TrichotomyAP;
 }
 
+void Constraint::printProofTree(std::ostream& out, size_t depth) const
+{
+#if IS_PROOFS_BUILD
```

Comment:
Maybe it would be better to also check if we are in debug and that the relevant debug/trace flag was turned on, as you do in some functions below. Otherwise an un-careful call to this function may lead to slower running times on debug (because of the loop inside).

Reply:
Interesting idea. I figured that the caller is responsible for gating this on the correct debug/trace flag, since this function wouldn't know the identity of the correct debug/trace flag: this function just gets an ostream.